dfusion_rust_core/src/models/order.rs

```diff
@@ -1,12 +1,18 @@
 use byteorder::{BigEndian, WriteBytesExt};
+use graph::data::store::Entity;
 use serde_derive::{Deserialize};
-use web3::types::H256;
+use std::sync::Arc;
+use web3::types::{H256, U256, Log};
 
 use crate::models::{Serializable, RollingHashable, iter_hash};
 
+use super::util::*;
+
 #[derive(Debug, Clone, Deserialize, Eq, Ord, PartialEq, PartialOrd)]
 #[serde(rename_all = "camelCase")]
 pub struct Order {
+    pub slot: Option<U256>,
```

Comment:
Given that these can semantically only be either both Some or both None, I'd prefer if this was a single field. 

The field would be an optional of a struct with two fields. It could be called something like `batch_information` or something like that. One the database level it's fine to have the datatype flat (as in not two different type of records).

Reply:
but wouldn't that mean the graph query would need to be nested? 
e.g 
```
sellOrders ( 
  where: {
    batch_information {
      slot: 123
      slot_index: 2
    }
  }
}
```
which isnt possible yet?

Reply:
On the db level, order could still be flat (as in have 7 fields: slot, slot_index, accountId, buy/sell token, buy/sell amount).

Reply:
i understand now